pkg/sql/sem/tree/parse_string.go

```diff
@@ -53,6 +53,12 @@ func ParseAndRequireString(t *types.T, s string, ctx ParseTimeContext) (Datum, e
 		i, err := ParseDInt(s)
 		return NewDOid(*i), err
 	case types.StringFamily:
+		// If the string type specifies a limit we truncate to that limit:
+		//   'hello'::CHAR(2) -> 'he'
+		// This is true of all the string type variants.
+		if t.Width() > 0 && int(t.Width()) < len(s) {
```

Comment:
interesting that parsing and storing strings that are too long have different results.

```
otan=# create temp table a (a varchar(1));
CREATE TABLE

otan=# insert into a values ('123');
ERROR:  value too long for type character varying(1)
```

Should this be rune count like we do here: https://github.com/cockroachdb/cockroach/blob/0f1a1631ea298c9818e4cf8e760a063cd17a9c63/pkg/sql/sqlbase/column_type_properties.go#L54 ?

Reply:
Hmm. I copied this from PerformCast (string->string case). I think you're right though, postgres docs say that the limit is n "characters (not bytes)".